packages/broker/test/integration/plugins/websocket/WebsocketPlugin.test.ts

```diff
@@ -0,0 +1,86 @@
+import WebSocket from 'ws'
+import { Stream, StreamrClient } from 'streamr-client'
+import { waitForEvent } from 'streamr-test-utils'
+import { startTracker, Tracker } from 'streamr-network'
+import { Broker } from '../../../../src/broker'
+import { createClient, startBroker, fastPrivateKey, createTestStream, createQueue } from '../../../utils'
+
+const WEBSOCKET_PORT = 12400
+const LEGACY_WEBSOCKET_PORT = 12401
+const TRACKER_PORT = 12402
+const NETWORK_PORT = 12403
+
+const privateKey = fastPrivateKey()
+
+describe('Websocket plugin', () => {
+
+    let stream: Stream
+    let streamrClient: StreamrClient
+    let wsClient: WebSocket
+    let tracker: Tracker
+    let broker: Broker
+
+    const createWebsocketClient = async (action: string): Promise<WebSocket> => {
+        const client = new WebSocket(`ws://localhost:${WEBSOCKET_PORT}/streams/${encodeURIComponent(stream.id)}/${action}`)
+        await waitForEvent(client, 'open')
+        return client
+    }
+
+    beforeAll(async () => {
+        tracker = await startTracker({
+            id: 'tracker',
+            host: '127.0.0.1',
+            port: TRACKER_PORT,
+        })
+        broker = await startBroker({
+            name: 'broker',
+            privateKey,
+            networkPort: NETWORK_PORT,
+            trackerPort: TRACKER_PORT,
+            wsPort: LEGACY_WEBSOCKET_PORT,
+            extraPlugins: {
+                websocket: {
+                    port: WEBSOCKET_PORT,
+                    sslCertificate: null
+                }
+            }
+        })
+    })
+
+    afterAll(async () => {
+        await Promise.allSettled([
+            broker.close(),
+            tracker.stop()
+        ])
+    })
+
+    beforeEach(async () => {
+        streamrClient = createClient(LEGACY_WEBSOCKET_PORT, privateKey, {
+            autoDisconnect: false
+        })
+        stream = await createTestStream(streamrClient, module)
+    })
+
+    afterEach(async () => {
+        wsClient?.close(),
+        await streamrClient?.ensureDisconnected()
+    })
+
+    test('publish', async () => {
```

Comment:
```suggestion
    test('message published to websocket plugin is received by client of legacy websocket plugin', async () => {
```

Reply:
The feature which we are testing here is publishing to the websocket. If this test fails, we know instantly that the publish functionality doesn't work.  

The fact that we ensure the functionality by reading the data via legacy websocket subscription is an implementation detail.

Reply:
Counter-argument: would this test not fail if (a) legacy ws plugin subscription logic, or (b) client implementation of  `streamrClient.subscribe`, had a bug that prevented subscriptions from properly working? If this test would fail, would you then claim publishing functionality of (non-legacy) websocket plugin does not work?

I'll leave it up to you, there is merit ofc for having a short name that underlines a clear-cut purpose for the test.

Reply:
True, but in that case half of our tests and many `streamr-client` tests would maybe fail.